python/ray/train/trainer.py

```diff
@@ -124,7 +125,8 @@ def __init__(
                     "request a positive number of `GPU` in "
                     "`resources_per_worker.")
 
-        self._executor = BackendExecutor(
+        remote_executor = ray.remote(num_cpus=0)(BackendExecutor)
```

Comment:
We should force the `BackendExecutor` actor on the head node since we don't have any fault tolerance for it. We can use the `force_on_current_node` in `tune/utils/util.py` and move it to a common `util/ml_utils/utils.py`

Reply:
Good point!